### ISSUE TITLE
Rollup dependency resolution error on installation

### ISSUE THREAD

**Issue by lankke** [hangovergames/create-backend#1]
The following dependency resolution error was encountered while trying to checkout and install the backend and subsequent calls to `npm run build` fail due to rollup not being installed:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: @rollup/plugin-commonjs@23.0.7
npm ERR! Found: rollup@1.32.1
npm ERR! node_modules/rollup
npm ERR!   rollup@"*" from the root project
npm ERR!   peerOptional rollup@"^1.20.0||^2.0.0||^3.0.0" from @rollup/plugin-babel@6.0.3
npm ERR!   node_modules/@rollup/plugin-babel
npm ERR!     @rollup/plugin-babel@"*" from the root project
npm ERR!   5 more (@rollup/plugin-inject, @rollup/plugin-json, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peerOptional rollup@"^2.68.0||^3.0.0" from @rollup/plugin-commonjs@23.0.7
npm ERR! node_modules/@rollup/plugin-commonjs
npm ERR!   @rollup/plugin-commonjs@"*" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: rollup@3.7.5
npm ERR! node_modules/rollup
npm ERR!   peerOptional rollup@"^2.68.0||^3.0.0" from @rollup/plugin-commonjs@23.0.7
npm ERR!   node_modules/@rollup/plugin-commonjs
npm ERR!     @rollup/plugin-commonjs@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

```
Environment:

OS: MacOS Monterey
Node: 18.9.0
NPM: 9.2.0

**Comment by lankke** [comment]
According to the README found [here](https://www.npmjs.com/package/@rollup/plugin-commonjs), the minimum rollup version required is Rollup v2.68.0 for plugin-commonjs v23.0.7

**Comment by thejhh** [comment]
The problem here might be the NodeJS v18. 

We have some known dependency problems which relate to the fact that there is incorrect version definitions in the uglifyjs plugin. Newer NodeJS/NPM breaks because of these. This might be another issue though.

**Comment by thejhh** [comment]
Which NodeJS version are you using?

We have some known dependency problems which relate to the fact that there is incorrect version definitions in the uglifyjs plugin. Newer NodeJS/NPM breaks because of these. This might be another issue though.

Workaround could be to use v14 for installation and switch to v18 after that.

**Comment by lankke** [comment]
So the issue appears to be that the following dependencies can manage with rollup versions below 2.68 (unlike plugin-commonjs)

```
node_modules/rollup
  rollup@"*" from the root project
  peerOptional rollup@"^1.20.0||^2.0.0||^3.0.0" from @rollup/plugin-babel@6.0.3
  node_modules/@rollup/plugin-babel
    @rollup/plugin-babel@"*" from the root project
  peerOptional rollup@"^1.20.0||^2.0.0||^3.0.0" from @rollup/plugin-inject@5.0.3
  node_modules/@rollup/plugin-inject
    @rollup/plugin-inject@"*" from the root project
  peerOptional rollup@"^1.20.0||^2.0.0||^3.0.0" from @rollup/plugin-json@6.0.0
  node_modules/@rollup/plugin-json
    @rollup/plugin-json@"*" from the root project
  peerOptional rollup@"^1.20.0||^2.0.0||^3.0.0" from @rollup/plugin-replace@5.0.2
  node_modules/@rollup/plugin-replace
    @rollup/plugin-replace@"*" from the root project
  peer rollup@">=0.66.0 <2" from rollup-plugin-uglify@6.0.4
  node_modules/rollup-plugin-uglify
    rollup-plugin-uglify@"*" from the root project
  peerOptional rollup@"^1.20.0||^2.0.0||^3.0.0" from @rollup/pluginutils@5.0.2
  ```
  Some of the above are compatible with 2.68+ but I guess the uglify one cannot?

**Comment by thejhh** [comment]
Workaround could be to use Node v14 for installation and switch to v18 after that -- but of course needs to be fixed

**Comment by lankke** [comment]
I'll try using Node 14 for installation for now :). Thanks JHH :)

**Comment by thejhh** [comment]
Yeah, uglifyjs plugin is actualy compatible but it is so old that it has incorrectly defined compatiblity settings

**Comment by lankke** [comment]
Is it this dependency that should be looked at?

peer rollup@**">=0.66.0 <2"** from rollup-plugin-uglify@6.0.4
  node_modules/rollup-plugin-uglify
    rollup-plugin-uglify@"*" from the root project

**Comment by thejhh** [comment]
I'll create a fork of it once I get to my workstation

**Comment by thejhh** [comment]
I can confirm that this relates to Node v18. It doesn't fail with v14.

**Comment by lankke** [comment]
I had some issues trying to install node 14 on my Mac OS Monterey machine (via nvm). It seems the dev Gods do not want me to install this... at least today :)

**Comment by thejhh** [comment]
I have a fix now for v18. 

There's another issue #2, I'll update our templates to fix that also.

**Comment by thejhh** [comment]
> I had some issues trying to install node 14 on my Mac OS Monterey machine (via nvm). It seems the dev Gods do not want me to install this... at least today :)

Yeah, I installed it using NVM and it actually compiled it there. (v18 did not need compiling.) :)

**Comment by lankke** [comment]
> 

In order to install Node 14 on my Mac I needed to run this first 
```
arch -x86_64 zsh
nvm install 14
```
I guess the M1 Macs need to run Rosetta for node versions < 15

**Comment by thejhh** [comment]
> > 
> 
> In order to install Node 14 on my Mac I needed to run this first
> 
> ```
> arch -x86_64 zsh
> nvm install 14
> ```
> 
> I guess the M1 Macs need to run Rosetta for node versions < 15

No, I don't think so... 

```
$ file /Users/jhh/.nvm/versions/node/v18.12.1/bin/node
/Users/jhh/.nvm/versions/node/v18.12.1/bin/node: Mach-O 64-bit executable arm64
```

I have also it installed from the official site globally as v14.

**Comment by thejhh** [comment]
Did you have full compile system installed? That might be the case; I have full xcode toolset so it could find a C++ compiler and compile that natively :D

**Comment by lankke** [comment]
No. I was just thinking that I should installed a C/C++ compiler as it did not find one during the setup. I haven't been actively using Macs for development for about 6 years so I'm still a bit rusty with how OSX works these days

**Comment by thejhh** [comment]
I am almost done with the fixed version for this issue.

Btw you might be interested about his: https://github.com/heusalagroup/hg-dashboard -- it is a complete full stack project template. We don't have an installer for that yet though.

**Comment by thejhh** [comment]
This issue has been fixed with version v1.0.5:

https://www.npmjs.com/package/@heusalagroup/create-backend
